webhooks/virtualmachine/validation/scsi_controller.go

```diff
@@ -0,0 +1,259 @@
+// © Broadcom. All Rights Reserved.
+// The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+// SPDX-License-Identifier: Apache-2.0
+
+package validation
+
+import (
+	"fmt"
+
+	vmopv1 "github.com/vmware-tanzu/vm-operator/api/v1alpha5"
+	pkgctx "github.com/vmware-tanzu/vm-operator/pkg/context"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+const (
+	// Maximum number of SCSI controllers allowed per VM.
+	maxSCSIControllers = int32(4)
+
+	// Maximum number of devices per SCSI controller type.
+	// Note: The controller itself occupies one slot.
+	maxParaVirtualSCSISlots = 63 // 64 targets - 1 for controller
```

Comment:
Wonder should we move this in `pkg/util/cns.go` since these are used all over the places